### PR TITLE
Common k8s E2E for PRs and master

### PR DESCRIPTION
**Pull request by jackfrancis**
**What this PR does / why we need it**: A circleci reconfiguration that runs the same k8s e2e tests for both PRs and master merges.

**Release note**:

```
NONE
```

### CODE REVIEW

**Comment by jackfrancis** [comment]
Also added a k8s version 1.7 test so we have coverage for that supported release.

File: .circleci/config.yml [hunk 1]
```diff
@@ -232,30 +298,54 @@ workflows:
...
               ignore: master
-      - ginkgo-k8s-windows-e2e:
+      - k8s-1.7-release-e2e:
           requires:
             - pr-e2e-hold
           filters:
             branches:
               ignore: master
-      - ginkgo-k8s-hybrid-e2e:
+      - k8s-windows-e2e:
```
Comment (jchauncey) [inline .circleci/config.yml]: should we name this `k8s-windows-default-release-e2e`?

File: .circleci/config.yml [hunk 2]
```diff
@@ -232,30 +298,54 @@ workflows:
...
+          filters:
+            branches:
+              only: master
+      - k8s-1.7-release-e2e-master:
           requires:
             - test
           filters:
             branches:
               only: master
+      - k8s-windows-e2e-master:
```
Comment (jchauncey) [inline .circleci/config.yml]: same thing here `k8s-windows-default-release-e2e-master`?

File: .circleci/config.yml [hunk 3]
```diff
@@ -232,30 +298,54 @@ workflows:
...
+          requires:
+            - test
+          filters:
+            branches:
+              only: master
+      - k8s-hybrid-e2e-master:
```
Comment (jchauncey) [inline .circleci/config.yml]: `k8s-hybrid-default-release-e2e-master`?

File: .circleci/config.yml [hunk 4]
```diff
@@ -232,30 +298,54 @@ workflows:
...
-      - ginkgo-k8s-hybrid-e2e:
+      - k8s-windows-e2e:
           requires:
             - pr-e2e-hold
           filters:
             branches:
-              ignore: master              
-      - kubernetes-e2e:
+              ignore: master
+      - k8s-hybrid-e2e:
```
Comment (jchauncey) [inline .circleci/config.yml]: `k8s-hybrid-default-release-e2e`?

File: .circleci/config.yml [hunk 5]
```diff
@@ -198,7 +190,81 @@ jobs:
           no_output_timeout: "30m"
       - store_artifacts:
           path: /go/src/github.com/Azure/acs-engine/_logs
-  ginkgo-k8s-hybrid-e2e:
+  k8s-hybrid-e2e:
```
Comment (jchauncey) [inline .circleci/config.yml]: made comment about this name below

File: .circleci/config.yml [hunk 6]
```diff
@@ -198,7 +190,81 @@ jobs:
...
+          echo 'export CREATE_VNET=true' >> $BASH_ENV
+          echo 'export CLIENT_ID=${CLUSTER_SERVICE_PRINCIPAL_CLIENT_ID}' >> $BASH_ENV
+          echo 'export CLIENT_SECRET=${CLUSTER_SERVICE_PRINCIPAL_CLIENT_SECRET}' >> $BASH_ENV
+      - run:
+          name: ginkgo k8s e2e tests
+          command: make build-binary test-kubernetes
+          no_output_timeout: "30m"
+      - store_artifacts:
+          path: /go/src/github.com/Azure/acs-engine/_logs
+  k8s-windows-e2e-master:
```
Comment (jchauncey) [inline .circleci/config.yml]: made comment about htis name below

File: .circleci/config.yml [hunk 7]
```diff
@@ -198,7 +190,81 @@ jobs:
...
+          echo 'export CLUSTER_DEFINITION=examples/windows/kubernetes.json' >> $BASH_ENV
+          echo 'export CLIENT_ID=${CLUSTER_SERVICE_PRINCIPAL_CLIENT_ID}' >> $BASH_ENV
+          echo 'export CLIENT_SECRET=${CLUSTER_SERVICE_PRINCIPAL_CLIENT_SECRET}' >> $BASH_ENV
+      - run:
+          name: ginkgo k8s windows e2e tests
+          command: make build-binary test-kubernetes
+          no_output_timeout: "30m"
+      - store_artifacts:
+          path: /go/src/github.com/Azure/acs-engine/_logs
+  k8s-hybrid-e2e-master:
```
Comment (jchauncey) [inline .circleci/config.yml]: made comment about this name below

**Comment by jchauncey** [comment]
after we settle on the names for these jobs lets make sure we update the required checks section in the settings tab

**Comment by jackfrancis** [comment]
While we're discussing names, I'd actually prefer to just abandon "default" in favor of explicit versions. The default-->explicit version transformation is something that we can test well w/ unit tests. What we really want to verify w/ E2E is that as the codebase evolves, we don't add regressions to any specific supported versions.

File: .circleci/config.yml [hunk 8]
```diff
@@ -180,7 +153,26 @@ jobs:
...
+  k8s-1.7-release-e2e:
+    working_directory: /go/src/github.com/Azure/acs-engine
+    docker:
+      - image: quay.io/deis/go-dev:v1.2.0
+        environment:
+          GOPATH: /go
+    steps:
+      - checkout
+      - run: |
+          echo 'export CLUSTER_DEFINITION=examples/e2e-tests/kubernetes/release/1.7/definition.json' >> $BASH_ENV
```
Comment (jchauncey) [inline .circleci/config.yml]: So you dont have to copy these files. Instead you can set `ORCHESTRATOR_RELEASE` to whatever version you want (1.7, 1.8, etc...) and it splice that into the definition before it runs generate. This means we just change the circle configuration and we can test any version we like.

**Comment by jackfrancis** [comment]
@jchauncey please check latest commit, thanks!

File: .circleci/config.yml [hunk 9]
```diff
@@ -199,6 +202,7 @@ jobs:
     steps:
       - checkout
       - run: |
+          echo 'export ORCHESTRATOR_RELEASE=1.7'
```
Comment (jchauncey) [inline .circleci/config.yml]: do you need to do `>> $BASH_ENV` here?

**Comment by jchauncey** [comment]
looks good other than my last comment.

**Pull request merged**